src/types/post.ts

```diff
@@ -22,3 +22,12 @@ export const PostDataSchema: ZodType<PostData[]> = z.array(
     post_id: z.string(),
```

Comment:
Projede zod varsa type'in aynisindan zod schema uretecegine tersten gidebilirsin soyle:
```
const PostDataSchema = z.object({
  title: z.string(),
  summary: z.string(),
  content: z.string(),
  category: z.string(),
  image: z.string(),
  post_id: z.string(),
});

const PostDataArraySchema = z.array(PostDataSchema)
type PostData = z.infer<typeof PostDataSchema>;
```

PostData gokten inmis de sanki elinle tek tek yazmisin gibi tum propertyleri doldurur otomatik
typescript cok feci bisi